log_forwarded_for_ip/__init__.py

```diff
@@ -3,13 +3,25 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
 from werkzeug.serving import WSGIRequestHandler
+from openerp import models
 
 
+class log_forwarded_for_ip_installed(models.AbstractModel):
+    _name = 'log.forwarded.for.ip.installed'
+
+address_string_original = WSGIRequestHandler.address_string
+
 def address_string(self):
-    forwarded_for = self.headers.get('X-Forwarded-For', '').split(',')
-    if forwarded_for and forwarded_for[0]:
-        return forwarded_for[0]
-    else:
+    if self.env.get('log.forwarded.for.ip.installed'):
```

Comment:
You can replace it by
```python
if module_name in self.env.registry._init_modules:
```
no more abstract model to use
Reference:

https://github.com/OCA/connector/issues/85
https://github.com/OCA/connector/blob/8.0/connector/connector.py#L64